src/java.base/share/classes/java/text/NumberFormat.java

```diff
@@ -883,9 +883,11 @@ public Object clone() {
     }
 
     /**
-     * Returns true if grouping is used in this format. For example, in the
-     * English locale, with grouping on, the number 1234567 might be formatted
-     * as "1,234,567". The grouping separator as well as the size of each group
+     * Returns true if grouping is used in this format. This applies to both
+     * formatting and parsing. For example, in the English locale, with grouping on,
+     * the number 1234567 might be formatted as "1,234,567". For the same format
+     * with grouping off, the String "1,234,567" might be parsed as 1.
```

Comment:
Although you used `might`, it may be helpful to mention the leniency here. I think it will throw exception in the strict mode.

Reply:
I originally included a strict behavior example as well but reverted it because the default behavior for NumberFormat does not support strict/non-strict. But I think we can include it as long as the wording is careful.

For the formatting example, I inferred that the "might" wording was regarding that there is no guarantee (however unlikely) that `,` remains the grouping separator and a size of 3 for the English locale. That's why I included that wording in the (previous) parsing example.

I changed the wording so that we can remove the ambiguous "might" wording. https://github.com/openjdk/jdk/pull/23813/commits/8c4ff2eb8c1746b524b661f37296cea6586161d9 contains these updates.